frontend/src/v5/validation/schemes.ts

```diff
@@ -17,85 +17,91 @@
-					}))
-	)),
-	defaultView: Yup.string()
-		.nullable()
-		.transform((value) => (value === EMPTY_VIEW._id ? null : value)),
-	latitude: numberField.required(),
-	longitude: numberField.required(),
-	angleFromNorth: numberField
-		.min(0,
-			formatMessage({
-				id: 'federations.angle.error.min',
-				defaultMessage: 'Angle cannot be smaller than 0',
-			}))
-		.max(360,
-			formatMessage({
-				id: 'federations.angle.error.max',
-				defaultMessage: 'Angle cannot be greater than 360',
-			}))
-		.transform((value) => value ?? 0),
-	x: numberField.required(),
-	y: numberField.required(),
-	z: numberField.required(),
-});
+	return Yup.object().shape({
+		name: Yup.string()
+			.min(2,
+				formatMessage({
+					id: 'settings.name.error.min',
+					defaultMessage: 'Name must be at least 2 characters',
+				}))
+			.max(120,
+				formatMessage({
+					id: 'settings.name.error.max',
+					defaultMessage: 'Name is limited to 120 characters',
+				}))
+			.required(
+				formatMessage({
+					id: 'settings.name.error.required',
+					defaultMessage: 'Name is a required field',
+				}),
+			),
+		desc: Yup.lazy((value) => (
+			value === ''
+				? Yup.string().strip()
+				: Yup.string()
+					.min(1,
+						formatMessage({
+							id: 'settings.desc.error.min',
+							defaultMessage: 'Description must be at least 1 character',
+						}))
+					.max(600,
+						formatMessage({
+							id: 'settings.desc.error.max',
+							defaultMessage: 'Description is limited to 600 characters',
+						}))
+		)),
+		unit: Yup.string().required().default('mm'),
+		code: Yup.lazy((value) => (
+			value === ''
+				? Yup.string().strip()
+				: Yup.string()
+					.min(1,
+						formatMessage({
+							id: 'settings.code.error.min',
+							defaultMessage: 'Code must be at least 1 character',
+						}))
+					.max(50,
+						formatMessage({
+							id: 'settings.code.error.max',
+							defaultMessage: 'Code is limited to 50 characters',
+						}))
+					.matches(/^[\w|_|-]*$/,
+						formatMessage({
+							id: 'settings.code.error.characters',
+							defaultMessage: 'Code can only consist of letters and numbers',
+						}))
+		)),
+		...(isContainer ? { type: Yup.string() } : {}),
+		defaultView: Yup.string()
+			.nullable()
+			.transform((value) => (value === EMPTY_VIEW._id ? null : value)),
+		latitude: numberField.required(),
+		longitude: numberField.required(),
+		angleFromNorth: numberField
+			.min(0,
+				formatMessage({
+					id: 'settings.angle.error.min',
+					defaultMessage: 'Angle cannot be smaller than 0',
+				}))
+			.max(360,
+				formatMessage({
+					id: 'settings.angle.error.max',
+					defaultMessage: 'Angle cannot be greater than 360',
+				}))
+			.transform((value) => value ?? 0),
+		x: numberField.required(),
+		y: numberField.required(),
+		z: numberField.required(),
+	});
+};
+
+export const FederationSettingsSchema = SettingsSchema();
+export const ContainerSettingsSchema = SettingsSchema(true);
```

Comment:
Instead of having SettingsSchema as a function I think we can keep it like the shape schema and extend it. I feel that is easier to see the difference between the schemas, making it a bit more readble.

```suggestion
export const FederationSettingsSchema = SettingsSchema;
export const ContainerSettingsSchema =SettingsSchema .shape({ type: Yup.string() });
```